src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php

```diff
@@ -38,15 +39,18 @@ class DumpDataCollector extends DataCollector implements DataDumperInterface
     private $requestStack;
     private $dumper;
     private $dumperIsInjected;
+    private $serverDumper;
+    private $sourceContextProvider;
 
-    public function __construct(Stopwatch $stopwatch = null, $fileLinkFormat = null, string $charset = null, RequestStack $requestStack = null, DataDumperInterface $dumper = null)
+    public function __construct(Stopwatch $stopwatch = null, $fileLinkFormat = null, string $charset = null, RequestStack $requestStack = null, DataDumperInterface $dumper = null, ServerDumper $serverDumper = null)
```

Comment:
why do we need a new argument? can't we pass the ServerDumper instance using $dumper?
this would require the serverdumper to accept a fallback dumper as argument to be used when the server is down
would allow better design than isServerUp, WDYT?

Reply:
Honestly I don't remember enough all the implications of these changes. What I remember is that I did them to keep the exact current behavior when not enabling the server dumper. I tried again, but I'm unable to get it right by using `$dumper`. 
I'd appreciate if you want to give it a try yourself (playing with both `server_dump` enabled/disabled) but I hope it won't be a blocker as the current behavior looks satisfying to me. Thanks!